### PR TITLE
ADEN-5630 Disable 3x3 and 2x2 sizes on pages with FV

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/ADEN-5630

### CODE REVIEW

File: extensions/wikia/AdEngine/js/provider/gpt/adSizeFilter.js [hunk 1]
```diff
@@ -19,41 +21,59 @@ define('ext.wikia.adEngine.provider.gpt.adSizeFilter', [
...
 
 	function getNewSizes(sizes, maxWidth, fallbackSizes) {
-		var goodSizes = [];
-
-		for (var i = 0; i < sizes.length; i += 1) {
-			if (sizes[i][0] <= maxWidth) {
-				goodSizes.push(sizes[i]);
-			}
-		}
+		var goodSizes = (sizes || []).filter(function(size) {
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/provider/gpt/adSizeFilter.js:23]: Such modern :poodle:

File: extensions/wikia/AdEngine/js/spec/provider/gpt/adSizeFilter.spec.js [hunk 2]
```diff
@@ -39,66 +42,80 @@ describe('ext.wikia.adEngine.provider.gpt.adSizeFilter', function () {
...
 
 		expect(getModule().filter('foo', sizesIn)).toEqual(sizesOut);
 	});
 
 	it('Returns sizes unmodified for TOP_LEADERBOARD (and variants) for large screens', function () {
 		spyOn(mocks, 'getDocumentWidth').and.returnValue(2000);
+		spyOn(mocks.adSizeConverter, 'toArray').and.returnValue([[728, 90], [1030, 130], [970, 365], [980, 150]]);
 
-		var sizesIn = [[728, 90], [1030, 130], [970, 365], [980, 150]],
+		var sizesIn = '728x90,1030x130,970x365,980x150',
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/spec/provider/gpt/adSizeFilter.spec.js]: Can we add here `3x3` and see if the new logic works (if it gets removed)?

File: extensions/wikia/AdEngine/js/spec/provider/gpt/adSizeFilter.spec.js [hunk 3]
```diff
@@ -140,9 +160,10 @@ describe('ext.wikia.adEngine.provider.gpt.adSizeFilter', function () {
 	it('Filter 728x90 size for PREFOOTER_LEFT_BOXAD for small screens' +
 		' when override prefooters feature enabled', function () {
 		spyOn(mocks, 'getContentWidth').and.returnValue(1000);
+		spyOn(mocks.adSizeConverter, 'toArray').and.returnValue([[728, 90], [468, 60], [300, 250]]);
 		mocks.overridePrefootersSizes = true;
 
-		var sizesIn = [[728, 90], [468, 60], [300, 250]],
+		var sizesIn = '728x90,468x60,300x250',
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/spec/provider/gpt/adSizeFilter.spec.js]: Let's also add one more test case for `MOBILE_TOP_LEADERBOARD` and `2x2` getting removed :japanese_ogre:

File: extensions/wikia/AdEngine/js/provider/gpt/adSizeFilter.js [hunk 4]
```diff
@@ -19,41 +21,59 @@ define('ext.wikia.adEngine.provider.gpt.adSizeFilter', [
...
+				return size[0] <= maxWidth;
+			});
 
 		log(['getNewSizes', 'sizes: ', sizes, 'goodSizes: ', goodSizes], 'debug', logGroup);
 
 		return goodSizes.length ? goodSizes : fallbackSizes;
 	}
 
+	function removeSize(slotSizes, sizeToRemove) {
+		var adContext = win.Mercury ?
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/provider/gpt/adSizeFilter.js]: `getAdContext()` ?

File: extensions/wikia/AdEngine/js/provider/gpt/adElement.js [hunk 5]
```diff
@@ -21,9 +20,9 @@ define('ext.wikia.adEngine.provider.gpt.adElement', [
 			this.node.id = this.id;
 		}
 		if (slotTargeting.size) {
-			this.sizes = adSizeConverter.convert(slotTargeting.size);
-			this.sizes = adSizeFilter.filter(slotName, this.sizes);
+			this.sizes = adSizeFilter.filter(slotName, slotTargeting.size);
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/provider/gpt/adElement.js]: IMO it's cleaner to filter and convert it separately. Right now your `adSizeFilter` knows and do too much

File: extensions/wikia/AdEngine/js/provider/gpt/adSizeFilter.js [hunk 6]
```diff
@@ -19,41 +21,59 @@ define('ext.wikia.adEngine.provider.gpt.adSizeFilter', [
...
+				win.Mercury.Modules.Ads.getInstance().adsContext :
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/provider/gpt/adSizeFilter.js]: Is it good idea to use Mercury modules even if it will be in app?

Comment (dianafa) [reply extensions/wikia/AdEngine/js/provider/gpt/adSizeFilter.js]: nothing better comes to my mind in this case

File: extensions/wikia/AdEngine/js/provider/gpt/adElement.js (hunk 5 above)
Comment (dianafa) [reply extensions/wikia/AdEngine/js/provider/gpt/adElement.js]: I would generally agree, but in this case it's' much more convenient to convert it inside `filterSizes` method. For some of the filtering usecases it's easier to operate on string, whereas for other it's more convenient to iterate over array, that's why conversion of the format is done in such way. I'll try to split those operations for bigger readability though

**Review by v3rth — APPROVED** [review]

**Pull request merged**